classes/Product.php

```diff
@@ -3379,6 +3379,43 @@ public static function priceCalculation(
             $real_quantity
         );
 
+        // If cart has been ordered the price is saved in OrderDetails
+        // When we want only the reduction no need for this
+        if ($id_cart && !$only_reduc) {
+            $orderId = Order::getIdByCartId($id_cart);
+            if (false !== $orderId) {
+                $sql = new DbQuery();
+                $sql->select('od.*, t.rate AS tax_rate');
+                $sql->from('order_detail', 'od');
+                $sql->where('od.`id_order` = ' . (int) $orderId);
+                $sql->where('od.`product_id` = ' . (int) $id_product);
+                if (Combination::isFeatureActive()) {
+                    $sql->where('od.`product_attribute_id` = ' . (int) $id_product_attribute);
+                }
+                $sql->leftJoin('order_detail_tax', 'odt', 'odt.id_order_detail = od.id_order_detail');
+                $sql->leftJoin('tax', 't', 't.id_tax = odt.id_tax');
+                $res = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($sql);
```

Comment:
Is there any particular reason why you're not using `->getRow` here?

Reply:
and... it is breaking change right?

Reply:
why not `getRow` ? just lack of knowledge of the Db possibilities ^^ I just copied a query and adapted it, I'm gonna test with getRow

About the BC break, you're right this is a big change in Cart/Price computations!! Do you think it should be a togglable feature (by adding a parameter) Or do we assume this is the new expected behaviour always and document this BC break?

@eternoendless @matks  I'd like to have your opinions on this as well?

Reply:
I'm afraid we need additional parameter 😒

Reply:
If we add a new parameter (like `$useOrderPrices` or `$freezeOrderPrices`) then this parameter will need to be added, at least, in:
- `Product::priceCalculation`
- `Product::getPriceStatic`
- `Cart::getProducts`
- `Cart::updateQty`
- `Cart::getOrderTotal`
- `CartRule::autoAddToCart`

That's not an exhaustive list as, potentially, all the function that call the listed ones would also need to have this extra parameter, and the ones that call them, and the ones that call the ones that call them 😅

So I don't know if the extra parameter would be the wisest choice considering the amount of code that needs to be modified, and the potential dependencies that also need updates.

Also one easier alternative could be to add a new configuration that would enable/disabled this feature?

Reply:
That said it would be nice if we had this option parameter 🤔. Because it would allow us to decide if it should be applied or not on the fly (like via a checkbox in the Order page or something)

Reply:
that's sound like a little workaround, you could combine Configuration::get and Configuration::set to do this, is this what you have in mind?

Reply:
That was a suggestion yes, but as you said it's more a workaround so i don't think it's a very good idea I'm gonna try and add this extra parameter in a dedicated commit just so that we can get an idea of the impact

Reply:
Another idea is to create completely separated method, I'm pretty sure we don't need everything from original one 🤔 what do you think?

Reply:
from what I see you're working only on values from order detail, can't we just based on that?

Reply:
you set price very early with `self::$_prices[$cache_id] = $price;` - you don't use most of code which is above and below yours, there's no use of specific price, address etc.

yeah, this is my suggestion, let's create a separated method, maybe even in `OrderDetailCore` to retrieve prices informations related to specific order